src/components/user-defined-sets/CurrentSetManager.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import Sets from './sets';
+
+const CURRENT_SELECTION = 'Current selection';
+
+export default class CurrentSetManager extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isEditing: false,
+      setName: '',
+    };
+
+    this.startEditing = this.startEditing.bind(this);
+    this.stopEditing = this.stopEditing.bind(this);
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  startEditing(event) {
+    event.preventDefault();
+    const { sets } = this.props;
+    const nextIndex = Array.from(sets.namedSets.keys()).length + 1;
```

Comment:
Could you just use `.size`? https://immutable-js.github.io/immutable-js/docs/#/OrderedMap/size

```suggestion
    const nextIndex = sets.namedSets.size + 1;
```